raster/r.surf.fractal/testsuite/test_r_surf_fractal.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+
+"""
+MODULE:    Test of r.surf.fractal
+
+AUTHOR(S): Shashank Shekhar Singh <shashankshekharsingh1205 gmail com>
+
+PURPOSE:   Test fractal surface generation with r.surf.fractal module
+
+COPYRIGHT: (C) 2024 by Shashank Shekhar Singh and the GRASS Development Team
+
+This program is free software under the GNU General Public
+License (>=v2). Read the file COPYING that comes with GRASS
+for details.
+"""
+
+import os
+
+import grass.script as gs
+
+from grass.gunittest.case import TestCase
+from grass.gunittest.main import test
+
+
+class FractalTestCase(TestCase):
+    """Test case for r.surf.fractal"""
+
+    # Raster map name to be used as output
+    output = "fractal_result"
+
+    @classmethod
+    def setUpClass(cls):
+        """Set up necessary environment"""
+        # Set up temporary computational region
+        cls.use_temp_region()
+        # Only 100,000,000 seem to reasonably (not 100%) ensure that all values
+        # are generated, so exceeding of ranges actually shows up.
```

Comment:
What kind of ranges you are referring to? I didn't see any range check. Usually a tiny computational region suffices to save on CI execution time.

Reply:
Would you mind helping me with the test suite please, because I haven't previously worked with them and therefore I was struggling with how to write them.

Thanks @marisn

Reply:
See for example test for r.random.cells and [test documentation](https://grass.osgeo.org/grass83/manuals/libpython/gunittest.html). Run r.surf.fractal first with certain fixed seed and save the raster statistics (computed with r.univar) for reference in the test. You can then use assertRasterFitsUnivar to ensure the test-generated raster with the same fixed seed will always give the same values. This type of test is not necessarily testing the output is correct, but it helps detect any regressions in the tool in the future and also that with a fixed seed you always get the same results.